compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -208,7 +208,8 @@ class CompilationTests extends ParallelTesting {
       defaultOutputDir + "lib/src/:" +
       // as well as bootstrapped compiler:
       defaultOutputDir + "dotty1/dotty1/:" +
-      Jars.dottyInterfaces
+      Jars.dottyInterfaces,
+      "-Ycheck-reentrant"
```

Comment:
@felixmulder could you please check if this is the right place to put the check?